Refactoring code, add location and added twitter and facebook in the StationDTO.

pablogrela opened the pull request:
Refactoring code, add location and added twitter and facebook in the StationDTO.
 - Code Formatting
 - Remove deprecated methods
 - Change if`s to switch
 - Add location intent to google maps
- The facebook and twitter url is added to the StationDTO, that is to say the information is recieved by the       request https://cuacfm.org/radioco/api/1/radiocom/radiostation.
- The text non elements textView has been changed to make it look better.
- Change the String baseURL to https://cuacfm.org/radioco/api/1/radiocom/

pablogrela committed:
Se ha refactoriazado el codigo de home, también se quitaron los métodos deprecated.
Se ha separada la url base, a una de programmes y otra de station
Se ha cambiado la url de los programas a https://cuacfm.org/radioco/programmes/json/
Se ha cambiado el textView de no elements para que tuviese una mejor apariencia

pablogrela committed:
Se cambio la url de los programas de cuacfm a:
api/1/radiocom/programmes
Siguiendo las indicaciones del creador de radioco.

pablogrela committed:
Se ha vuelto a poner una url en vez de dos, por petición del creador, despues se vifurca en programmes y radiostation

pablogrela committed:
Now the information of twitter and facebook is obtained from the StationDTO, which receives the information of the url provided.
If no information is provided on the StationDTO, the information provided in the GlobalValues is collected.

pablogrela committed:
Added the position of the radio station.
The length and latitude are taken from the StationDTO.

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -302,27 +310,38 @@ public void charguePodcast() {
         processFragment(podcastFragment, mContext.getString(R.string.action_podcast));
     }
 
+    public void chargueMap() {
```
This method must be named loadMap

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -60,13 +63,15 @@
     private SharedPreferences prefs;
     private SharedPreferences.Editor edit;
     private static final int REQUEST_WRITE_STORAGE = 112;
+    private static final int MY_PERMISSIONS_REQUEST_READ_CONTACTS = 112;
```
Why do you need to request read contacts?

pablogrela replied:
For nothing.
I was testing the permissions issue for the mapFragment but in the end I did not use it and that variable escaped me.
I delete it.

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -302,27 +310,38 @@ public void charguePodcast() {
         }
     }
 
     public void chargueFacebook() {
```
Rename this method to loadFacebook

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -302,27 +310,38 @@ public void charguePodcast() {
+    }
+
 
     public void chargueTwitter() {
```
Rename this method to loadTwitter

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -234,7 +230,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
         // automatically handle clicks on the Home/Up button, so long
         // as you specify a parent activity in AndroidManifest.xml.
-        int id = item.getItemId();
+        //int id = item.getItemId();
```
remove the line if its not used

pablogrela replied:
ok

ficiverson reviewed (commented):
addDrawerListener

ficiverson commented:
Please make also the following changes:

-Translate all spanish comment to english
-Rename the following methods like this:
chargueEmisora->loadStation
chargueGaleria->loadGallery
chargueEmsision->loadNews
charguePodcast->loadPodcast
chargueMap->loadMap
chargueTwitter->loadTwitter
chargueFacebook->loadFacebook
ProcessBuilder->processBuilder
chargueDeveloper->loadDeveloperInfo

It was a great job!! After this minor changes I will merge PR

pablogrela committed:
-Rename the following methods like this:
chargueEmisora->loadStation
chargueGaleria->loadGallery
chargueEmsision->loadNews
charguePodcast->loadPodcast
chargueMap->loadMap
chargueTwitter->loadTwitter
chargueFacebook->loadFacebook
ProcessBuilder->processBuilder
chargueDeveloper->loadDeveloperInfo

In loadMap, has put the two types of intent, one to put a mark on the map and oto that sets the route from the source where the user.

pablogrela commented:
Done
